examples/pytorch/metapath2vec/reading_data.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+
+np.random.seed(12345)
+
+class DataReader:
+    NEGATIVE_TABLE_SIZE = 1e8
+
+    def __init__(self, inputFileName, min_count, care_type):
+
+        self.negatives = []
+        self.discards = []
+        self.negpos = 0
+        self.care_type = care_type
+        self.word2id = dict()
+        self.id2word = dict()
+        self.sentences_count = 0
+        self.token_count = 0
+        self.word_frequency = dict()
+
+        self.inputFileName = inputFileName
+        self.read_words(min_count)
+        self.initTableNegatives()
+        self.initTableDiscards()
+
+    def read_words(self, min_count):
+        word_frequency = dict()
+        for line in open(self.inputFileName, encoding="utf8"):
+            line = line.split()
+            if len(line) > 1:
+                self.sentences_count += 1
+                for word in line:
+                    if len(word) > 0:
+                        self.token_count += 1
+                        word_frequency[word] = word_frequency.get(word, 0) + 1
+
+                        if self.token_count % 1000000 == 0:
+                            print("Read " + str(int(self.token_count / 1000000)) + "M words.")
+
+        wid = 0
+        #index2type = {}
+        for w, c in word_frequency.items():
+            if c < min_count:
+                continue
+            self.word2id[w] = wid
+            self.id2word[wid] = w
+            #index2type[wid] = w[0]
+            self.word_frequency[wid] = c
+            wid += 1
+        #type2indices = {}
+        #all_types = set(index2type.values())
+        #for node_type in all_types:
+        #    type2indices[node_type] = []
+
+        #for node_index, node_type in index2type.items():
+        #    type2indices[node_type].append(node_index)
+
+        #for node_type in all_types:
+        #    type2indices[node_type] = np.array(type2indices[node_type])
+
+        self.word_count = len(self.word2id)
+        #self.index2type = index2type
+        #self.type2indices = type2indices
+        print("Total embeddings: " + str(len(self.word2id)))
+
+    def initTableDiscards(self):
+        t = 0.0001
+        f = np.array(list(self.word_frequency.values())) / self.token_count
+        self.discards = np.sqrt(t / f) + (t / f)
```

Comment:
it seems this isn't used. Why do you have this function?

Reply:
This function is called in the very front, preparing a frequency table for sub-sampling.